lms/djangoapps/certificates/migrations/0025_update_generatedcertificate_managers.py

```diff
@@ -0,0 +1,44 @@
+# Generated by Django 2.2.20 on 2021-04-27 14:59
+
+from django.db import migrations
+import django.db.models.manager
+import lms.djangoapps.certificates.models
+
+
+class Migration(migrations.Migration):
+    """
+    Clean records in the `CERTIFICATES_GENERATEDCERTIFICATE` table that are in the downloadable state but also have
+    old errors still part of their certificate record.
+
+    As part of this migration we are also altering the Managers for the GeneratedCertificate model. We need the ability
+    to access the `objects` attribute for the cleanup.
+    """
+    def cleanup_certificate_records(apps, schema_editor):
+        GeneratedCertificate = apps.get_model('certificates', 'GeneratedCertificate')
+
+        cert_list = (
+            GeneratedCertificate.eligible_certificates
+            .filter(status='downloadable')
+            .exclude(error_reason='')
+        )
+
+        for cert in cert_list:
+            print(f'I am updating cert for {cert.user.username} with status {cert.status} and error of {cert.error_reason}')
+            cert.error_reason = ''
+            cert.save()
+
+    dependencies = [
+        ('certificates', '0024_delete_allowlistgenerationconfiguration'),
+    ]
+
+    operations = [
+        migrations.AlterModelManagers(
+            name='generatedcertificate',
+            managers=[
+                ('objects', django.db.models.manager.Manager()),
+                ('eligible_certificates', lms.djangoapps.certificates.models.EligibleCertificateManager()),
+                ('eligible_available_certificates', lms.djangoapps.certificates.models.EligibleAvailableCertificateManager()),
+            ],
+        ),
+        migrations.RunPython(cleanup_certificate_records, reverse_code=migrations.RunPython.noop)
```

Comment:
I tried to keep these two migrations separate. I tried to do the Manager migration first, then follow up with the data migration. Django didn't like this. I would get the following error:
```
CommandError: Conflicting migrations detected; multiple leaf nodes in the migration graph: (0026_cleanup_generatedcertificate_errors, 0025_update_generatedcertificate_managers in certificates).
To fix them run 'python manage.py makemigrations --merge'
```

I was able to fix this by combining the two migrations into one.